doc/api/sqlite.md

```diff
@@ -115,6 +115,26 @@ added: v22.5.0
 
 Constructs a new `DatabaseSync` instance.
 
+### `database.backup(destination[, options])`
```

Comment:
Not something that needs to happen in this PR necessarily, but we will need to rename the `DatabaseSync` class now that an async operation is being added. My plan was always to keep the sync and async APIs separated, similar to the fs module. Since we are breaking from that, I'll leave it up to others to decide on naming, keeping in mind that there is already a feature request for an async API.

Reply:
Yeah, it could be a start for that async API. 

https://github.com/nodejs/node/issues/54307

I asked about this in the issue but I didn't get an answer

Reply:
Question, @cjihrig . In your plans, would DatabaseSync have a backup feature?

Personally, I have no problem moving this to a `Database` class (without the `Sync`) as a start for that async API.

Reply:
Does it need to be a method on `DatabaseSync`? Could it be a separate API exposed on the `node:sqlite` module? e.g.

```js
const { backup, DatabaseSync } = require('node:sqlite');

const db = new DatabaseSync(...);
// ...
await backup(db);
```

Reply:
I like this idea!

Reply:
I pushed a change implementing that. Let's see how it goes 🙏